dist/sizzle.js

```diff
@@ -465,26 +465,51 @@ function createButtonPseudo( type ) {
  * @param {Boolean} disabled true for :disabled; false for :enabled
  */
 function createDisabledPseudo( disabled ) {
-	// Known :disabled false positives:
-	// IE: *[disabled]:not(button, input, select, textarea, optgroup, option, menuitem, fieldset)
-	// not IE: fieldset[disabled] > legend:nth-of-type(n+2) :can-disable
+
+	// Known :disabled false positives: fieldset[disabled] > legend:nth-of-type(n+2) :can-disable
 	return function( elem ) {
 
-		// Check form elements and option elements for explicit disabling
-		return "label" in elem && elem.disabled === disabled ||
-			"form" in elem && elem.disabled === disabled ||
+		// Only certain elements can match :enabled or :disabled
+		// https://html.spec.whatwg.org/multipage/scripting.html#selector-enabled
+		// https://html.spec.whatwg.org/multipage/scripting.html#selector-disabled
+		if ( "form" in elem ) {
+
+			// Check for inherited disabledness on relevant non-disabled elements:
+			// * listed form-associated elements in a disabled fieldset
+			//   https://html.spec.whatwg.org/multipage/forms.html#category-listed
+			//   https://html.spec.whatwg.org/multipage/forms.html#concept-fe-disabled
+			// * option elements in a disabled optgroup
+			//   https://html.spec.whatwg.org/multipage/forms.html#concept-option-disabled
+			// All such elements have a "form" property.
+			if ( elem.parentNode && elem.disabled === false ) {
+				return "label" in elem ?
```

Comment:
> That's how parser code usually looks like (with `if..else` not with ternaries, which is easy to understand to me at least), essentially this part is a parser but working with different data type - not strings but DOM, never saw a parser that love ternaries like that, in many modern languages, for example like go or rust ternary do not exist.

Go has named return values, and Rust has if-expressions. This is Javascript, where we have neither. Languages are different.

> And from my experience, when you write code with simple `if..else` you usually find ways to move out some parts of scope to helper functions which decreases cyclomatic complexity or easily find a ways to simplify and optimize things but never with ternaries.

I had already done that with the ternaries, which as I pointed out are exactly equivalent to the statements, since the result is always a single equality comparison expression. Which brings me to...

> > I'd rather optimize for size

> That's unclear, there is no evidence to it, from writing code for core with replacing such constructions, I can say that uglify was always outputed same gzip says as with ternaries.

As it turns out, this function is now in a form where Uglify can produce the _exact_ nested ternaries that I want, so I've implemented the statement-based version.

> Because you writing code like this and still using `jshint` and `jscs` it prevents us from enabling correct full check for dist version of jQuery - https://github.com/jquery/jquery/blob/master/dist/.eslintrc#L8-L17, currently, there is 10 rules we had to disable because of it.

I'm blocked on handling of expression parentheses (https://github.com/eslint/eslint/issues/4689#issuecomment-210123742 , https://github.com/eslint/eslint/issues/5270 ). Sizzle has lots of assignments inside statement conditions, and we use tight parentheses to indicate their intentionality.

> In any case if you had objections to it, i think you had to bring it up beforehand - https://github.com/jquery/eslint-config-jquery/issues/2 I `cc` everyone to that ticket

Like @dmethvin [said](https://github.com/jquery/jquery/pull/3233#discussion_r70829217), the consequences weren't obvious. I for one am frustrated by the separation of variable declarations from initializations that was required in jQuery core to satisfy `no-nested-ternary`.